### PR TITLE
Vamp Shapeshift/DNAScrambler Fixes & Desnowflakes Preferences Setup

### PR DESCRIPTION
**Pull request by KasparoVy**
:cl:
fix: Fixes a bug with Vampire Shapeshift where you didn't get a name appropriate to your species and clarifies the ability description.
fix: Fixes a bug with DNA scramblers whereby you wouldn't get a name appropriate to your species.
/:cl:

Ensures you won't be taking on another species using this vampire shapeshift by circumventing an exploit.
Desnowflakes preferences_setup.dm like I had done with preferences.dm in #7019.

### CODE REVIEW

File: code/modules/mob/new_player/preferences_setup.dm [hunk 1]
```diff
@@ -12,38 +17,38 @@
 	underwear = random_underwear(gender, species)
 	undershirt = random_undershirt(gender, species)
 	socks = random_socks(gender, species)
-	if(species == "Vulpkanin")
+	if(body_accessory_by_species[species] || check_rights(R_ADMIN, 0, usr))
```
Comment (tigercat2000) [inline code/modules/mob/new_player/preferences_setup.dm]: This could definitely cause hilarious bugs

Comment (KasparoVy) [inline code/modules/mob/new_player/preferences_setup.dm]: :rofl:. Clearly not hilarious enough to keep. A shame!

**Comment by KasparoVy** [comment]
Anything I've missed..?

**Comment by KasparoVy** [comment]
@tigercat2000 Still LGTY?

**Comment by KasparoVy** [comment]
@tigercat2000 Still LGTY?

:neutral_face:

**Comment by KasparoVy** [comment]
:wave:

**Comment by Fox-McCloud** [comment]
Merge conflicts. Sorry about them, but the species bitflag to traits one was...well, considering your work with stuff, I think you can understand why it took priority.

File: code/modules/mob/new_player/preferences_setup.dm [hunk 2]
```diff
@@ -1,8 +1,13 @@
...
 	//The mob should have a gender you want before running this proc. Will run fine without H
 /datum/preferences/proc/random_character(gender_override)
+	var/datum/species/S = all_species[species]
+	if(!istype(S)) //The species was invalid. Set the species to the default, fetch the datum for that species and generate a random character.
+		species = initial(species)
+		S = all_species[species]
 	var/datum/robolimb/robohead
-	if(species == "Machine")
+
+	if(S.flags & ALL_RPARTS)
```
Comment (Fox-McCloud) [inline code/modules/mob/new_player/preferences_setup.dm]: this is a bodyflag now

File: code/modules/mob/new_player/preferences_setup.dm [hunk 3]
```diff
@@ -12,38 +17,38 @@
...
 		m_styles["body"] = random_marking_style("body", species)
 		var/list/colours = randomize_skin_color(1)
 		m_colours["body"] = rgb(colours["red"], colours["green"], colours["blue"])
-	if(species in list("Vox", "Vulpkanin")) //Species with tail markings.
+	if(S.bodyflags & HAS_TAIL_MARKINGS) //Species with tail markings.
 		m_styles["tail"] = random_marking_style("tail", species, null, body_accessory)
 		var/list/colours = randomize_skin_color(1)
 		m_colours["tail"] = rgb(colours["red"], colours["green"], colours["blue"])
-	if(species != "Machine")
+	if(!(S.flags & ALL_RPARTS))
```
Comment (Fox-McCloud) [inline code/modules/mob/new_player/preferences_setup.dm]: Same here.

**Comment by KasparoVy** [comment]
Conflicts resolved.

**Comment by KasparoVy** [comment]
@Fox-McCloud Conflicts resolved. Thanks for notifying, totally gave up watching this

**Comment by Fox-McCloud** [comment]
@tigercat2000

**Pull request merged**